lk2nd/device/menu/menu.c

```diff
@@ -123,7 +124,10 @@ static uint16_t wait_key(void)
 
 static void opt_continue(void)   { cmd_continue(NULL, NULL, 0); }
 static void opt_reboot(void)     { reboot_device(0); }
-static void opt_recovery(void)   { reboot_device(RECOVERY_MODE); }
+static void opt_recovery(void)   {
+	boot_into_recovery = 1;
+	cmd_continue(NULL, NULL, 0);
+}
```

Comment:
I'd prefer this to be confined to the function scope
```suggestion
static void opt_recovery(void)
{
	extern unsigned boot_into_recovery;

	boot_into_recovery = 1;
	cmd_continue(NULL, NULL, 0);
}
```